File: src/main/java/akkastudy/closures/java/Closures.java
package akkastudy.closures.java;

import java.util.function.Function;

public class Closures {
    public static Integer foo(Function<Integer, Integer> f) {
        return f.apply(5);
    }

    public static void main(String[] args) {
        Integer x = 3;
        Function<Integer, Integer> y = (Integer z) -> x + z;
        System.out.println(foo(y));
    }
}

File: src/main/scala/akkastudy/closures/scala/Closures.scala
package akkastudy.closures.scala

object Closures {
  def foo(f: Int => Int): Int = f(5)

  def main(args: Array[String]) {
    val x: Integer = 3
    val add3: Int => Int = z => x + z
    System.out.println(foo(add3))
  }
}
